sdv/cag/fixed_increments.py

```diff
@@ -0,0 +1,218 @@
+
+            metadata (sdv.metadata.Metadata):
+                The input Metadata to use to validate the data.
+
+        Returns:
+            None
+        """
+        valid = self._check_if_divisible(
+            data, self._get_single_table_name(metadata), self.column_name, self.increment_value
+        )
+        if not valid.all():
+            invalid_rows_str = _get_invalid_rows(valid)
+            raise PatternNotMetError(
+                'The fixed increments requirement has been met because the data is not '
+                f"evenly divisible by '{self.increment_value}' for row indices: "
+                f'[{invalid_rows_str}]'
+            )
+
+    def _get_updated_metadata(self, metadata):
+        """Get the updated metadata after applying the pattern to the metadata.
+
+        Args:
+            metadata (sdv.metadata.Metadata):
+                The input Metadata to apply the pattern to.
+
+        Returns:
+            (sdv.metadata.Metadata): The updated Metadata with the pattern applied.
+        """
+        table_name = self._get_single_table_name(metadata)
+        original_columns = list(metadata.tables[table_name].columns)
+        increments_column = _create_unique_name(
+            self._fixed_increments_column_name, metadata.tables[table_name].columns.keys()
+        )
+        metadata = metadata.to_dict()
+        metadata['tables'][table_name]['columns'][increments_column] = {'sdtype': 'numerical'}
+        metadata = _remove_columns_from_metadata(
+            metadata,
+            table_name,
+            columns_to_drop=original_columns,
+        )
+        return metadata
+
+    def _fit(self, data, metadata):
+        """Learn the dtype of the column.
+
+        Args:
+            data (dict[pd.DataFrame]):
+                The data.
+
+            metadata (sdv.metadata.Metadata):
+                The input Metadata.
+        """
+        table_name = self._get_single_table_name(metadata)
+        self._dtype = data[table_name][self.column_name].dtype
+
+    def _check_if_divisible(self, data, table_name, column_name, increment_value):
+        """Check if a column is divisible by a given increment value.
+
+        Args:
+            data (dict[pd.DataFrame]):
+                The data.
+
+            table_name (str):
+                Name of the table.
+
+            column_name (str):
+                Name of the table to check divisibility.
+
+            increment_value (int):
+                the number with which divisibility needs to be checked.
+        """
+        isnan = pd.isna(data[table_name][column_name])
+        is_divisible = data[table_name][column_name] % increment_value == 0
+        return isnan | is_divisible
+
+    def _is_valid(self, data):
+        """Determine if the data is evenly divisible by the increment.
+
+        Args:
+            data (dict[pd.DataFrame]):
+                The data.
+
+        Returns:
+            pandas.Series:
```

Comment:
It's a dict, not a series

Reply:
Fixed:
https://github.com/sdv-dev/SDV/pull/2409/commits/ab7486461cdacaa4a32ab55016ba3204f780c406#diff-442749eb6608ea424c56131f8b5c2633ab037f4f758317361cc150224d2d4620R166